### ISSUE TITLE
How to handle namespaced packages like ruamel.yaml?

### ISSUE THREAD

**Issue by ericdill** [ericdill/depfinder#22]
From @ocefpaf in #21:

@ericdill and @tonyfast any advice on how to deal with named spaced packages like `ruamel.yaml`? `depfinder` will use the import name `ruamel` but there is no such package. However, the `ruamel.yaml.__package__` info is correct. I wonder if we could take advantage of that instead and produce a more accurate result.

@tonyfast responded:

The yaml question is interesting. Currently, depfinder is looking at the ast and doesn't know anything about the run time. We could import the modules so they are in sys.modules and look at the __package__ information.

**Comment by ericdill** [comment]
Hi @ocefpaf,

@tonyfast is totally correct. This code knows nothing about your runtime, so even if ruaml.yaml.__package__ exists, I don't think that we necessarily want depfinder to be aware of it.

[`get_imported_libs`](https://github.com/ericdill/depfinder/blob/master/depfinder/main.py#L192-L225) is the "simple" entry point that you can use to check how `depfinder` is going to handle a given import line.

```
In [2]: from depfinder import main

In [3]: main.get_imported_libs('import ruaml.yaml')
Out[3]: ImportCatcher: {'required': {'ruaml'}}
```

This get_imported_libs code makes use of the ImportFinder class to look for all the `import foo` and `from foo import bar` statements in your python code. There are two places where we would need to change this behavior to handle namespaced packages:

[visit_Import](https://github.com/ericdill/depfinder/blob/master/depfinder/main.py#L121) and two places in the `ImportFrom` function, [here](https://github.com/ericdill/depfinder/blob/master/depfinder/main.py#L146) and [here](https://github.com/ericdill/depfinder/blob/master/depfinder/main.py#L142).

Assuming that y'all agree on this, I think we want to do the following:

1. Have lines 121, 142 and 146 flow through a single function. 
2. Inside of that as-yet-unwritten-function, we could have another lookup that takes the full import string and checks it against a known list of namespaced content.
3. Move this namespaced list,  `_PACKAGE_MAPPING` and `_FAKE_PACKAGES` into a data file and update the code in depfinder to pull from that data file

**Comment by ericdill** [comment]
Closed by #23
